snippets/mongocompat/test.js

```diff
@@ -22,6 +22,49 @@ assert.strictEqual(minLong.bottom, 0);
 assert.strictEqual(minLong.exactValueString, "-9223372036854775808");
 const nl2 = NumberLong("200");
 assert.strictEqual(maxLong.compare(nl2), 1);
+
 const decimal = NumberDecimal("1.1");
 assert.strictEqual(decimal.toString(), 'NumberDecimal("1.1")');
 assert.strictEqual(decimal.tojson(), 'NumberDecimal("1.1")');
+
+const ts1 = Timestamp();
+assert.strictEqual(ts1.toString(), 'Timestamp(0, 0)');
+const ts2 = Timestamp(100, 200);
+assert.strictEqual(ts2.toString(), 'Timestamp(100, 200)');
+const ts3 = Timestamp(1.9, 2.1);
+assert.strictEqual(ts3.toString(), 'Timestamp(1, 2)');
+try {
+    Timestamp(-1, 0);
+    assert.fail('Should throw for negative time');
+} catch (e) {
+    assert(e.message.includes('must be non-negative'));
+}
+try {
+    Timestamp(0, 5000000000);
+    assert.fail('Should throw for i > uint32 max');
+} catch (e) {
+    assert(e.message.includes('not greater than 4294967295'));
+}
+const ts4 = Timestamp(123, 456);
```

Comment:
You'll also want to assert that this value `instanceof Timestamp`.

`instanceof` is generally a bad choice in JS most of the time, but since we're modifying the prototype chains here, an assertion that validates that the current behavior is kept intact does seem to make sense to me